src/ui/public/directives/file_upload.js

```diff
@@ -1,32 +1,80 @@
+import _ from 'lodash';
 import $ from 'jquery';
 import uiModules from 'ui/modules';
 var module = uiModules.get('kibana');
 
-module.directive('fileUpload', function ($parse) {
+var html = '<span class="dropzone" ng-transclude></span>';
+
+module.directive('fileUpload', function () {
   return {
-    restrict: 'A',
+    restrict: 'E',
+    transclude: true,
+    scope: {
+      onRead: '&',
+      onLocate: '&',
+      label: '='
+    },
+    template: html,
     link: function ($scope, $elem, attrs) {
-      var onUpload = $parse(attrs.fileUpload);
+      var $button = $elem.find('.upload');
```

Comment:
I guess you're right, it is a bit odd... my goal was to give the transcluded content full control over how the clickable element is displayed. Not sure how else to do it at the moment but I'll give it some thought as well.